File: Assets/AltUnityTester/AltUnityDriver/Notifications/BaseNotificationCallBacks.cs
using System;
using Altom.AltUnityDriver.Logging;

namespace Altom.AltUnityDriver.Notifications
{
    public class BaseNotificationCallBacks : INotificationCallbacks
    {
        private static readonly NLog.Logger logger = DriverLogManager.Instance.GetCurrentClassLogger();
        public void SceneLoadedCallback(AltUnityLoadSceneNotificationResultParams altUnityLoadSceneNotificationResultParams)
        {
            logger.Log(NLog.LogLevel.Info, String.Format("Scene {0} was loaded {1}", altUnityLoadSceneNotificationResultParams.sceneName, altUnityLoadSceneNotificationResultParams.loadSceneMode.ToString()));
        }
        public void SceneUnloadedCallback(string sceneName)
        {
            logger.Log(NLog.LogLevel.Info, String.Format("Scene {0} was unloaded", sceneName));
        }
        public void LogCallback(AltUnityLogNotificationResultParams altUnityLogNotificationResultParams)
        {
            logger.Log(NLog.LogLevel.Info, String.Format("Log of type {0} with message {1} was received", altUnityLogNotificationResultParams.level, altUnityLogNotificationResultParams.message));
        }
        public void ApplicationPausedCallback(bool applicationPaused)
        {
            logger.Log(NLog.LogLevel.Info, String.Format("Application paused: {0}", applicationPaused));
        }
    }
}

File: Bindings~/java/src/main/java/ro/altom/altunitytester/Notifications/BaseNotificationCallbacks.java
package ro.altom.altunitytester.Notifications;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ro.altom.altunitytester.Logging.AltUnityLogLevel;

public class BaseNotificationCallbacks implements INotificationCallbacks {
    protected static final Logger logger = LogManager.getLogger(BaseNotificationCallbacks.class);

    @Override
    public void SceneLoadedCallBack(
            AltUnityLoadSceneNotificationResultParams altUnityLoadSceneNotificationResultParams) {
        logger.info("Scene " + altUnityLoadSceneNotificationResultParams.sceneName + " was loaded "
                + altUnityLoadSceneNotificationResultParams.loadSceneMode);
    }

    @Override
    public void SceneUnloadedCallBack(String sceneName) {
        logger.info("Scene " + sceneName + " was unloaded ");
    }
    
    @Override
    public void LogCallBack(AltUnityLogNotificationResultParams altUnityLogNotificationResultParams) {
        logger.info("Log of type " + AltUnityLogLevel.values()[altUnityLogNotificationResultParams.level] + " with message " +
        altUnityLogNotificationResultParams.message + " was received");
    }

    @Override
    public void ApplicationPausedCallBack(boolean applicationPaused) {
        logger.info("Application paused: " + applicationPaused);
    }
}